runners_manager/vm_creation/openstack.py

```diff
@@ -38,40 +43,117 @@ def __init__(self, project_name, token, username, password, region):
+        self.cinder_client = cinderclient.client.Client('3', session=session, region_name=region)
 
-    def script_init_runner(self, name: str, token: int, vm_type: VmType,
-                           github_organization: str, installer: str):
+    @staticmethod
+    def script_init_runner(runner: Runner, token: int,
+                           github_organization: str, installer: str, need_new_volume: bool,
+                           template_name='init_runner_script_replace.sh'):
         file_loader = FileSystemLoader('templates')
         env = Environment(loader=file_loader)
         env.trim_blocks = True
         env.lstrip_blocks = True
         env.rstrip_blocks = True
-        template = env.get_template('init_runner_script.sh')
-        output = template.render(image=vm_type.image.lower(), installer=installer,
+
+        template = env.get_template(template_name)
+        output = template.render(installer=installer, create_volume=need_new_volume,
                                  github_organization=github_organization,
-                                 token=token, name=name, tags=','.join(vm_type.tags))
+                                 token=token, name=runner.name, tags=','.join(runner.vm_type.tags),
+                                 group='default')
+        print(output)
         return output
 
-    def create_vm(self, name: str, runner_token: int, vm_type: VmType,
-                  github_organization: str, installer: str):
+    def volume_logic(self, instance: novaclient.v2.servers.Server,
+                     volume_id: str or None, mount_point='/dev/vdb'):
+        if volume_id is None:
+            volume = self.cinder_client.volumes.create(
+                name=instance.name,
+                description='volume for runner manager',
+                volume_type="SATA",
+                size=5
+            )
+            volume_id = volume.id
+
+        else:
+            volume = self.cinder_client.volumes.get(volume_id)
+
+        while self.cinder_client.volumes.get(volume.id).status != "available":
+            print(f"wait available {self.cinder_client.volumes.get(volume.id).status}")
+            time.sleep(2)
+
+        self.cinder_client.volumes.attach(volume.id, instance.id, mount_point, mode='rw')
+        self.nova_client.volumes.create_server_volume(instance.id, volume.id, mount_point)
+        while self.cinder_client.volumes.get(volume.id).status != "in-use":
+            print(f"wait in use {self.cinder_client.volumes.get(volume.id).status}")
+            time.sleep(2)
+
+        return volume_id
+
+    def create_vm(self, runner: Runner, runner_token: int or None,
+                  github_organization: str, installer: str, need_new_volume=False):
+
+        sec_group_id = self.neutron.list_security_groups()['security_groups'][0]['id']
+        nic = {'net-id': self.neutron.list_networks(name='tenantnetwork1')['networks'][0]['id']}
```

Comment:
I don't know the exact answer here, but those settings seems to be specific to our openstack platform.

Could you please add a comment, like a TODO to set those as configs